playbooks/replicate.yml

```diff
@@ -0,0 +1,128 @@
+      ansible.builtin.command: /usr/bin/rsync -avz "{{ ansible_user }}@{{ ansible_hostname }}:{{ media_backup_path }}" "{{ dest_backup_dir }}" -e "ssh -o StrictHostKeyChecking=no"
+      # become: true
+      # become_user: "{{ deploy_user }}"
+      delegate_to: "{{ dest_host }}"
+
+    - name: Make sure unzipped database dump does not exist (if it does, bunzip2 will fail)
+      ansible.builtin.file:
+        path: "{{ dest_backup_path_unzipped }}"
+        state: absent
+      delegate_to: "{{ dest_host }}"
+
+    # postgres module can create zipped dump file, but at least on OSX
+    # it fails to unzip when attempting to restore;
+    # use command line since builtin unarchive task does not handle bz2
+    - name: Unzip dumped database on {{ dest_host }}
+      ansible.builtin.command: bunzip2 "{{ dest_backup_path }}"
+      delegate_to: "{{ dest_host }}"
+
+    # if we don't clear the database first, loading the dump results in
+    # errors/warnings about constraints that already exist
+    # NOTE: in local setup, remove database fails if django is running (client connected to db)
+    - name: Remove the database "{{ application_db_name }}" on target host {{ dest_host }} if it exists
+      community.postgresql.postgresql_db:
+        name: "{{ application_db_name }}"
+        login_host: '{{ dest_postgres_host }}'
+        login_user: '{{ application_dbuser_name }}'
+        login_password: '{{ application_dbuser_password }}'
+        state: "absent"
+      delegate_to: "{{ dest_host }}"
+
+    - name: Create empty database "{{ application_db_name }}" on target host {{ dest_host }}
+      community.postgresql.postgresql_db:
+        name: "{{ application_db_name }}"
+        login_host: '{{ dest_postgres_host }}'
+        login_user: '{{ application_dbuser_name }}'
+        login_password: '{{ application_dbuser_password }}'
+        state: "present"
+        owner: '{{ application_dbuser_name }}'
+      delegate_to: "{{ dest_host }}"
+
+    - name: Load the database "{{ application_db_name }}" on target host {{ dest_host }}
+      community.postgresql.postgresql_db:
+        name: "{{ application_db_name }}"
+        encoding: 'UTF-8'
+        login_host: '{{ dest_postgres_host }}'
+        login_user: '{{ application_dbuser_name }}'
+        login_password: '{{ application_dbuser_password }}'
+        owner: '{{ application_dbuser_name }}'
+        target: "{{ dest_backup_path_unzipped }}"
+        state: "restore"
+      # become: true
+      # become_user: postgres
+      delegate_to: "{{ dest_host }}"
+      # async: 1800
+      # poll: 60
+
+    - name: Extract media to destination media directory
+      ansible.builtin.unarchive:
+        src: "{{ dest_media_backup_path }}"
+        dest: "{{ dest_media_dir }}"
+      delegate_to: "{{ dest_host }}"
+
+
+# could django migration be done as a second playbook? use existing role/task
+- hosts: localhost    # local won't actually work without some setup...
```

Comment:
If you always want to run the migration tasks when you run the db backup/restore, you can probably use `import_tasks` here. 

`- import_tasks: ../../roles/django/tasks/migrate.yml`

See https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse.html#re-using-files-and-roles for more details.